### PR TITLE
docs: add user spec to swagger

### PR DESCRIPTION
**Pull request by jonnyguio**
Refs #1992

### CODE REVIEW

File: docs/reference/api.yaml [hunk 1]
```diff
@@ -274,6 +275,408 @@ paths:
...
+          description: Not found
+          schema:
+            $ref: '#/definitions/ErrorMessage'
+      tags: 
+        - users
+      security:
+        - Bearer: []
+  /1.0/users/api-key:
+    get:
+      operationId: APITokenShow
```
Comment (andrestc) [inline docs/reference/api.yaml]: `APITokenGet` looks more common for an API.

File: docs/reference/api.yaml [hunk 2]
```diff
@@ -274,6 +275,408 @@ paths:
...
+      operationId: UserSSHKeyAdd
+      description: Add SSH key to logged user.
+      parameters:
+        - name: UserSSHKeyAddData
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/UserSSHKeyAddData'
+      consumes:
+        - x-www-form-url-encoded
```
Comment (andrestc) [inline docs/reference/api.yaml]: `application/json`

File: docs/reference/api.yaml [hunk 3]
```diff
@@ -274,6 +275,408 @@ paths:
...
+          description: Not found
+          schema:
+            $ref: '#/definitions/ErrorMessage'
+      tags:
+        - users
+      security:
+        - Bearer: []
+  /1.0/users/info:
+    get:
+      operationId: UserInfo
```
Comment (andrestc) [inline docs/reference/api.yaml]: I think I'd use `UserGet` (ignoring the `/info` on the endpoint)

**Comment by jonnyguio** [comment]
In my opinion, it's important to be as clear as possible what the context of the operation. Quota, for instance, is also a component of the app, so "QuotaChange" wouldn't be clear if it's about changing an user quota or an app quota. 

On the other hand, as there are no other usage of ssh keys, I think that the User part could be left out.

File: docs/reference/api.yaml [hunk 4]
```diff
@@ -274,6 +274,412 @@ paths:
...
+      parameters:
+        - name: email
+          in: query
+          type: string
+          required: true
+      produces: 
+        - application/json
+      responses:
+        '200': 
+          description: <API_TOKEN>
```
Comment (andrestc) [inline docs/reference/api.yaml]: Maybe just: `API Token.`

File: docs/reference/api.yaml [hunk 5]
```diff
@@ -274,6 +274,412 @@ paths:
...
+      parameters:
+        - name: email
+          in: query
+          type: string
+          required: true
+      produces:
+        - application/json
+      responses:
+        '200': 
+          description: <API_TOKEN>
```
Comment (andrestc) [inline docs/reference/api.yaml]: Same as above

File: docs/reference/api.yaml [hunk 6]
```diff
@@ -274,6 +274,412 @@ paths:
...
+            $ref: '#/definitions/ErrorMessage'
+      tags:
+        - users
+      security:
+        - Bearer: []
+    post:
+      operationId: SSHKeyAdd
+      description: Add SSH key to logged user.
+      parameters:
+        - name: UserSSHKeyAddData
```
Comment (andrestc) [inline docs/reference/api.yaml]: Just `SSHKeyAddData`?

File: docs/reference/api.yaml [hunk 7]
```diff
@@ -274,6 +274,412 @@ paths:
...
+    get:
+      operationId: UserGet
+      description: Get information on logged user.
+      produces:
+        - application/json
+      responses:
+        '200': 
+          description: OK
+          schema:
+            $ref: '#/definitions/UserInfoResponse'
```
Comment (andrestc) [inline docs/reference/api.yaml]: UserGetResponse

File: docs/reference/api.yaml [hunk 8]
```diff
@@ -274,6 +274,412 @@ paths:
...
+          description: User e-mail.
+      consumes:
+        - application/x-www-form-urlencoded
+      produces:
+        - application/json
+      responses:
+        '200': 
+          description: OK
+          schema:
+            $ref: '#/definitions/UserQuotaViewResponse'
```
Comment (andrestc) [inline docs/reference/api.yaml]: `UserQuotaGetResponse`

File: docs/reference/api.yaml [hunk 9]
```diff
@@ -274,6 +274,412 @@ paths:
...
+      description: Get quota of an user.
+      parameters:
+        - name: email
+          in: path
+          required: true
+          type: string
+          minLength: 1
+          description: User e-mail.
+      consumes:
+        - application/x-www-form-urlencoded
```
Comment (andrestc) [inline docs/reference/api.yaml]: I think we can drop the `consumes` here.

File: docs/reference/api.yaml [hunk 10]
```diff
@@ -274,6 +274,412 @@ paths:
...
+      operationId: ChangePassword
+      description: Change password of logged user.
+      parameters:
+        - name: ChangePasswordData
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/ChangePasswordData'
+      consumes:
+       - application/x-www-form-urlencoded
```
Comment (andrestc) [inline docs/reference/api.yaml]: `application/json`

**Pull request merged**